assertj-core/src/main/java/org/assertj/core/api/Assertions.java

```diff
@@ -1415,11 +1416,54 @@ public static Throwable catchThrowable(ThrowingCallable shouldRaiseThrowable) {
    * @see #catchThrowable(ThrowingCallable)
    * @since 3.9.0
    */
+  @Deprecated(since = "3.23.11")
```

Comment:
```suggestion
  @Deprecated
```

The `since` attribute is supported on Java 9+ but we are still targeting Java 8.

Reply:
Should I put that bit in the documentation then?

Reply:
This has been done, please resolve the conservation so the PR can move forward.